constraint-solver/src/range_constraint.rs

```diff
@@ -14,9 +14,11 @@ use powdr_number::{log2_exact, FieldElement, LargeInt};
 /// Note that the same constraint can have multiple representations.
 #[derive(Clone, PartialEq, Eq, PartialOrd, Ord, Debug, Hash)]
 pub struct RangeConstraint<T: FieldElement> {
-    /// Bit-mask.
+    /// Bit-mask. A value `x` is allowed only if `x & mask == x` (when seen as unsigned integer).
```

Comment:
```suggestion
    /// Bit-mask. A value `x` is allowed iff `x & mask == x` (when seen as unsigned integer).
```

?

Reply:
and in general, range constraints are always under-approximation, i.e. the real constraints could be tighter. Maybe I should add that in the docs.